Scripts/Runtime/Jobs/Collections/CollectionAccessControl.cs

```diff
@@ -0,0 +1,94 @@
+    public static class CollectionAccessControl<TContext, TKey>
+    {
+        private static readonly Dictionary<TKey, CollectionAccessController<TContext, TKey>> s_Controllers = new Dictionary<TKey, CollectionAccessController<TContext, TKey>>();
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void Init()
+        {
+            Dispose();
+        }
+        
+        /// <summary>
+        /// Returns an instance of an <see cref="CollectionAccessController{TContext, TKey}"/> for a given key.
+        /// Will create a new one if it doesn't already exist.
+        /// </summary>
+        /// <param name="key">The key to lookup or create the instance of the <see cref="CollectionAccessController{TContext, TKey}"/>.</param>
+        /// <returns>The <see cref="CollectionAccessController{TContext,TKey}"/> instance.</returns>
+        public static CollectionAccessController<TContext, TKey> GetOrCreate(TKey key = default)
+        {
+            if (!s_Controllers.TryGetValue(key, out CollectionAccessController<TContext, TKey> accessController))
+            {
+                accessController = new CollectionAccessController<TContext, TKey>(key);
+                accessController.OnDisposed += CollectionAccessController_OnDisposed;
+                s_Controllers.Add(key, accessController);
+            }
+
+            return accessController;
+        }
+
+        /// <summary>
+        /// Removes an instance of <see cref="CollectionAccessController{TContext, TKey}"/> for a given key from the
+        /// lookup. If successful, the instance will also be disposed.
+        /// </summary>
+        /// <param name="key">The key to lookup the instance of the <see cref="CollectionAccessController{TContext, TKey}"/></param>
+        /// <returns>True if successfully removed and disposed. False if the key was not found in the lookup.</returns>
+        public static bool RemoveAndDispose(TKey key = default)
+        {
+            if (!s_Controllers.TryGetValue(key, out CollectionAccessController<TContext, TKey> accessController))
+            {
+                return false;
+            }
+
+            s_Controllers.Remove(key);
+
+            if (accessController.IsDisposing || accessController.IsDisposed)
+            {
+                return true;
+            }
+            
+            accessController.OnDisposed -= CollectionAccessController_OnDisposed;
+            accessController.Dispose();
+
+            return true;
+        }
+        
+        /// <summary>
+        /// High level dispose to be used when exiting from an ECS/DOTS mode into another part of the application
+        /// that doesn't used ECS/DOTS. All controllers will be removed from the lookup and disposed.
+        /// </summary>
+        public static void Dispose()
```

Comment:
Maybe better to call this `RemoveAndDisposeAll()` since you're not really disposing the static class.

Reply:
Changed.